Source/WebCore/dom/Element.cpp

```diff
@@ -4091,9 +4093,20 @@ static Element* parentCrossingFrameBoundaries(const Element* element)
     return element->document().ownerElement();
 }
 
-void Element::webkitRequestFullscreen()
+void Element::webkitRequestFullscreen(RefPtr<DeferredPromise>&& promise)
 {
-    document().fullscreenManager().requestFullscreenForElement(*this, FullscreenManager::EnforceIFrameAllowFullscreenRequirement);
+    document().fullscreenManager().requestFullscreenForElement(*this, WTFMove(promise), FullscreenManager::EnforceIFrameAllowFullscreenRequirement);
+}
+
+// https://fullscreen.spec.whatwg.org/#dom-element-requestfullscreen
+// FIXME: Add proper support.
+void Element::requestFullscreen(FullscreenOptions&&, Ref<DeferredPromise>&& promise)
+{
+    if (!document().isFullyActive()) {
+        promise->reject(Exception { TypeError, "Document is not fully active"_s });
+        return;
+    }
+    webkitRequestFullscreen(WTFMove(promise));
 }
```

Comment:
I'd like `webkitRequestFullscreen` to be a wrapper around `requestFullscreen`, not the other way round. 

```
    if (!document().isFullyActive()) {
         promise->reject(Exception { TypeError, "Document is not fully active"_s });
         return;
     }
```
can move into `FullscreenManager::requestFullscreenForElement()` as a preflight check (for both APIs).

Another benefit is that `webkitRequestFullscreen` can avoid taking in the `promise` as parameter as well.